### PR TITLE
fixes #5897 - when admin updates a user, email is not required

### PR DESCRIPTION
**Pull request by unorthodoxgeek**

### CODE REVIEW

File: test/unit/user_test.rb [hunk 1]
```diff
@@ -17,8 +17,18 @@ def setup
...
     u = FactoryGirl.create(:user, :mail => nil)
-    u.firstname = 'Bob'
-    refute_valid u, :mail
+    as_user(u) do
+      u.firstname = 'Bob'
+      refute_valid u, :mail
+    end
+  end
+
+  test "mail is still opitonal when admin does the updating" do
```
Comment (lzap) [inline test/unit/user_test.rb]: Typo.

File: test/unit/user_test.rb [hunk 2]
```diff
@@ -17,8 +17,18 @@ def setup
...
+    as_admin do
+      u = FactoryGirl.create(:user, :mail => nil)
+      u.firstname = 'Bob'
+      assert_valid u
+    end
```
Comment (lzap) [inline test/unit/user_test.rb]: I think you need to call `save!` to trigger something. Or use `update_attributes!`.

File: test/unit/user_test.rb [hunk 3]
```diff
@@ -17,8 +17,18 @@ def setup
 
   test "should have mail when updating" do
     u = FactoryGirl.create(:user, :mail => nil)
-    u.firstname = 'Bob'
-    refute_valid u, :mail
+    as_user(u) do
+      u.firstname = 'Bob'
+      refute_valid u, :mail
```
Comment (lzap) [inline test/unit/user_test.rb]: Ditto.

**Comment by lzap** [comment]
Ok tested, works fine. I am not sure about these unit tests. Someone chip in please? I thought Rails validations are called when you try to save, but `u.firstname=` does not trigger that. Otherwise :+1:

File: test/unit/user_test.rb (hunk 2 above)
Comment (unorthodoxgeek) [inline test/unit/user_test.rb]: Nope. Assert valid triggers the valid? method. It runs validations

**Comment by lzap** [comment]
Ok `assert_valid` does that, I mis-read that. Please fix the typo and we are done here.

**Comment by domcleal** [comment]
Thinking about this further, it should apply to _any_ user that is editing this user, and not just admins.  Consider a "manager" type role with edit_users, the behaviour of not requiring the e-mail address should be the same as an admin.

I'm also unsure if we should permit _removal_ of an e-mail address.  If a user already has one set, should we only allow it to be changed and not removed entirely?

**Comment by unorthodoxgeek** [comment]
@domcleal  - updated, now user doesn't require email if its email is currently blank

**Comment by domcleal** [comment]
Merged as 564b322, thanks @unorthodoxgeek.

**Pull request closed without merging**